crates/nu-cli/src/commands/headers.rs

```diff
@@ -0,0 +1,75 @@
+use crate::commands::WholeStreamCommand;
+use crate::context::CommandRegistry;
+use crate::prelude::*;
+use futures::stream::StreamExt;
+use indexmap::IndexMap;
+use nu_errors::ShellError;
+use nu_protocol::Dictionary;
+use nu_protocol::{ReturnSuccess, Signature, UntaggedValue, Value};
+
+pub struct Headers;
+#[derive(Deserialize)]
+pub struct HeadersArgs {}
+
+impl WholeStreamCommand for Headers {
+    fn name(&self) -> &str {
+        "headers"
+    }
+    fn signature(&self) -> Signature {
+        Signature::build("headers")
+    }
+    fn usage(&self) -> &str {
+        "Use the first row of the table as headers"
+    }
+    fn run(
+        &self,
+        args: CommandArgs,
+        registry: &CommandRegistry,
+    ) -> Result<OutputStream, ShellError> {
+        args.process(registry, headers)?.run()
+    }
+}
+
+pub fn headers(
+    HeadersArgs {}: HeadersArgs,
+    RunnableContext { input, name, .. }: RunnableContext,
+) -> Result<OutputStream, ShellError> {
+    let stream = async_stream! {
+        let rows: Vec<Value> = input.values.collect().await;
+
+        //the headers are the first row in the table
+        let headers: Vec<String> = match &rows[0].value {
+            UntaggedValue::Row(d) => {
+                d.entries.iter().map(|(_, v)| {
+                    match v.as_string() {
+                        Ok(s) => s,
+                        Err(_) => String::from("empty-header") //If a cell that should contain a header name is empty, we need to fill it with something.
+                    }
+                }).collect()
+            }
+            _ => panic!("Could not find headers")
```

Comment:
As part of the clean-up to get this ready to be a full PR, you can change these to output Err(ShellError). You'll see the pattern in a lot of the other commands, so you should be able to find some good examples to copy and use.

Reply:
I changed it to the maybe a bit diffuse `UntaggedRuntimeError`. I believe that this error occurs when the table is empty and no headers can be found, and so I thought about UnexpectedEof, but I am not sure where I can take the span from. 

Thank you for your comment!